File: node/tests/schemas/jsonstream-blocks.ts
import { expect } from "@jest/globals";

export const schema = {
  chain: expect.any(Number),
  block_number: expect.any(String),
  hash: expect.any(String),
  parent_hash: expect.any(String),
  uncles_hash: expect.any(String),
  miner: expect.any(String),
  state_root: expect.any(String),
  transactions_root: expect.any(String),
  receipts_root: expect.any(String),
  gas_used: expect.any(String),
  gas_limit: expect.any(String),
  base_fee_per_gas: expect.any(String),
  extra_data: expect.any(String),
  logs_bloom: expect.any(String),
  timestamp: expect.any(Number),
  difficulty: expect.any(String),
  total_difficulty: expect.any(String),
  size: expect.any(String),
  mix_hash: expect.any(String),
  nonce: expect.any(String),
};

File: rust/tests/schemas/jsonstream-blocks.rs
use serde_json::json;
use serde_json::Value;

pub fn schema() -> Value {
    json!({
        "chain": Value::Number(serde_json::Number::from(0)),
        "block_number": Value::String(String::new()),
        "hash": Value::String(String::new()),
        "parent_hash": Value::String(String::new()),
        "uncles_hash": Value::String(String::new()),
        "miner": Value::String(String::new()),
        "state_root": Value::String(String::new()),
        "transactions_root": Value::String(String::new()),
        "receipts_root": Value::String(String::new()),
        "gas_used": Value::String(String::new()),
        "gas_limit": Value::String(String::new()),
        "base_fee_per_gas": Value::String(String::new()),
        "extra_data": Value::String(String::new()),
        "logs_bloom": Value::String(String::new()),
        "timestamp": Value::Number(serde_json::Number::from(0)),
        "difficulty": Value::String(String::new()),
        "total_difficulty": Value::String(String::new()),
        "size": Value::String(String::new()),
        "mix_hash": Value::String(String::new()),
        "nonce": Value::String(String::new())
    })
}
